controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -107,6 +108,14 @@ func (r *StorageConsumerReconciler) Reconcile(ctx context.Context, request recon
 		return reconcile.Result{}, err
 	}
 
+	if controllerutil.AddFinalizer(r.storageConsumer, StorageConsumerFinalizer) {
```

Comment:
who is going to remove this finalizer? isn't the PR addressing only internal client & consumer, adding here should be out of scope as the controller deals w/ both internal & external consumers.